users/service.go

```diff
@@ -45,21 +71,39 @@ type Service interface {
 	// other reason, non-nil error values are returned in response.
 	Identify(string) (string, error)
 
-	// Get authenticated user info for the given token.
+	// Get authenticated user info for the given token
 	UserInfo(ctx context.Context, token string) (User, error)
+
+	// GenerateResetToken email where mail will be sent.
+	// host is used for generating reset link.
+	GenerateResetToken(_ context.Context, email, host string) error
+
+	// UpdatePassword change users password in reset flow
+	UpdatePassword(_ context.Context, token, password string) error
+
+	//SendToken sends reset password link to email
+	SendToken(_ context.Context, host, email, token string) error
```

Comment:
Add open tracing support for these methods, or add an issue, so that we don't forget about it.